client/src/main/java/io/pravega/client/stream/impl/Controller.java

```diff
@@ -240,7 +241,7 @@
      * @param from The position from which to find the remaining bytes.
      * @return The total number of bytes beyond the provided positions.
      */
-    CompletableFuture<Set<Segment>> getSuccessors(StreamCut from);
+    CompletableFuture<Pair<Set<Segment>, String>> getSuccessors(StreamCut from);
```

Comment:
two points:
1. please update the java doc for return type.
2. this piggybacking of getSuccessors call to also return a new delegation token is a bit hacky. 
instead the client could make a getSuccessors and check if it needs to get new token in the meanwhile or not and perform authentication again if the existing token is about to expire.

Reply:
1. Updated.
2. The approach proposed in the PDP is that the calls to controller return a delegation token along with the requested resource info. It does not make sense to make another call to get the token (as this call again goes through authentication/authorization - which we have already done for the earlier call). See `Controller.proto` for more changes. I do not think it makes sense to change that approach. As discussed offline, using a `Pair` here definitely looks clumsy. Will replace this with a proper class.